File: backend-express/src/bin/testStorage.ts
import { storage } from "@/lib/clients";
import { getImagePath } from "@/static";

async function test() {
    const path = getImagePath("avatar1.jpg");
    const destination = await storage.uploadFile(path);
    const url = await storage.getSignedUrl(destination);
    console.log(url);
}

if (require.main === module) {
    test().then(() => console.log("Finished testing storage client"));
}

File: backend-gin/internal/bin/test_storage.go
package bin

import (
	"backend/internal/services/instances"
	"backend/internal/static"
	"context"
	"fmt"
)

func TestStorage() {
	// Get Storage client
	ctx := context.Background()
	path := static.GetImagePath("avatar1.jpg")
	storage := instances.GetStorage()
	defer storage.Close()

	// Test uploading a file
	destination, err := storage.UploadFile(ctx, path, "")
	if err != nil {
		fmt.Println(err.Error())
	}

	// Test a getting a signed URL
	url, err := storage.GetSignedURL(destination, 3600)
	if err != nil {
		fmt.Println(err)
	} else {
		fmt.Println(url)
	}
}
